core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java

```diff
@@ -399,11 +404,15 @@ public SmallRyeConfig initConfiguration(LaunchMode launchMode, Properties buildS
         for (ConfigClassWithPrefix mapping : getBuildTimeVisibleMappings()) {
             builder.withMapping(mapping.getKlass(), mapping.getPrefix());
         }
-        return builder.build();
+
+        builder.withInterceptors(buildConfigTracker);
+        var config = builder.build();
+        buildConfigTracker.configure(config);
```

Comment:
If you need to configure the interceptor, you can use a `ConfigSourceInterceptorFactory` instead. This provides you access to the context with the current config.

Reply:
Thanks for pointing that out. Given that I'd need to obtain the reference to the created interceptor, not sure whether it will end up being more elegant at the end though.